### PR TITLE
sync CAPZ maintainer team members with OWNERS

### PR DESCRIPTION
**Pull request by nojnhuh**
Syncing with recent changes made to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/OWNERS_ALIASES

cc: @sonasingh46

### CODE REVIEW

File: config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml [hunk 1]
```diff
@@ -122,6 +122,8 @@ teams:
     - jackfrancis
     - mboersma
     - shysank
+    - nojnhuh
+    - sonasingh46
```
Comment (nojnhuh) [inline config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml:126]: @CecileRobertMichon Should we also be added to `cluster-api-provider-azure-admins`?

Comment (nojnhuh) [reply config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml:126]: (pending above)
/hold

Comment (nojnhuh) [reply config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml]: Yep, will do.

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

File: config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml (hunk 1 above)
Comment (nojnhuh) [reply config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml:122]: /hold cancel

**Pull request merged**